src/probnum/problems/_problems.py

```diff
@@ -1,19 +1,19 @@
 """Definitions of problems currently solved by probabilistic numerical methods."""
 
 import dataclasses
-import typing
+from collections import abc
+from typing import Callable, Optional, Sequence, Union
 
 import numpy as np
 import scipy.sparse
 
-import probnum.linops as pnlo
-import probnum.type as pntp
-from probnum import randvars
+from probnum import linops, randvars
+from probnum.type import FloatArgType
 
 
 @dataclasses.dataclass
-class RegressionProblem:
-    r"""Regression problem.
+class TimeSeriesRegressionProblem:
```

Comment:
I think renaming this is a good idea, because (i) it is the argument type that is passed to Kalman filtering and particle filtering, and (ii) all the functions (e.g. merge_..., or filter()) rely on something iterable. This only works for time series